packer/builder/azure/arm/config_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"testing"
+)
+
+// List of configuration parameters that are required by the ARM builder.
+var requiredConfigValues = []string{
+	"capture_name_prefix",
+	"capture_container_name",
+	"client_id",
+	"client_secret",
+	"image_offer",
+	"image_publisher",
+	"image_sku",
+	"location",
+	"storage_account",
+	"subscription_id",
+	"tenant_id",
+}
+
+func TestConfigShouldProvideReasonableDefaultValues(t *testing.T) {
+	c, _, err := newConfig(getArmBuilderConfiguration(), getPackerConfiguration())
+
+	if err != nil {
+		t.Errorf("Expected configuration creation to succeed, but it failed!\n")
+		t.Fatalf(" errors: %s\n", err)
+	}
+
+	if c.UserName == "" {
+		t.Errorf("Expected 'UserName' to be populated, but it was empty!")
+	}
+
+	if c.VMSize == "" {
+		t.Errorf("Expected 'VMSize' to be populated, but it was empty!")
+	}
+}
+
+func TestConfigShouldDefaultVMSizeToStandardA1(t *testing.T) {
+	c, _, _ := newConfig(getArmBuilderConfiguration(), getPackerConfiguration())
+
+	if c.VMSize != "Standard_A1" {
+		t.Errorf("Expected 'VMSize' to default to 'Standard_A1', but got '%s'.", c.VMSize)
+	}
+}
+
+func TestUserShouldProvideRequiredValues(t *testing.T) {
+	builderValues := make(map[string]string)
+	var lastElement = requiredConfigValues[len(requiredConfigValues)-1]
+
+	for _, v := range requiredConfigValues {
+		builderValues[v] = "--ignored--"
+
+		_, _, err := newConfig(getArmBuilderConfigurationFromMap(builderValues), getPackerConfiguration())
+		if lastElement != v && err == nil {
```

Comment:
so this means that the first element is not checked wrt to it being required?
I think a better approach might be to just `getArmBuilderConfiguration()` and then delete a key from the map for each key in `requiredConfigValues` and assert that it errors appropriately?